### ISSUE TITLE
[REVIEW]: libRL: A Python library for the characterization of microwave absorption

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1868]
**Submitting author:** @1mikegrn (<a href="http://orcid.org/0000-0002-2525-1274">Michael Green</a>)
**Repository:** <a href="https://github.com/1mikegrn/libRL-package" target ="_blank">https://github.com/1mikegrn/libRL-package</a>
**Version:** v1.1.0
**Editor:** @dfm
**Reviewer:** @kjappelbaum, @mdoucet
**Archive:** <a href="https://doi.org/10.5281/zenodo.3555457" target="_blank">10.5281/zenodo.3555457</a>

## Status

[![status](https://joss.theoj.org/papers/61a4e825e28a34037d0ba03f335a7537/status.svg)](https://joss.theoj.org/papers/61a4e825e28a34037d0ba03f335a7537)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/61a4e825e28a34037d0ba03f335a7537"><img src="https://joss.theoj.org/papers/61a4e825e28a34037d0ba03f335a7537/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/61a4e825e28a34037d0ba03f335a7537/status.svg)](https://joss.theoj.org/papers/61a4e825e28a34037d0ba03f335a7537)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/1mikegrn/libRL-package" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@kjappelbaum & @mdoucet, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @dfm know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [X] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [X] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [X] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/GreenScarz/libRL-package">repository url</a>?
- [X] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [X] **Contribution and authorship:** Has the submitting author (@1mikegrn) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [X] **Installation:** Does installation proceed as outlined in the documentation?
- [X] **Functionality:** Have the functional claims of the software been confirmed?
- [X] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [X] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [X] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [X] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [X] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [X] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [X] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [X] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [X] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [X] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [X] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [X] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/GreenScarz/libRL-package">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@1mikegrn) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @kjappelbaum, @mdoucet it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01868/joss.01868/10.21105.joss.01868.pdf)

**Comment by 1mikegrn** [comment]
Hey @matthewturk, just hoping to get an update on some sort of timetable we can expect for conducting the peer-review. I'm getting ready to submit a set of high-impact research manuscripts and I'd like to be able to reference this manuscript therein for more coverage of the software. Thanks :)

**Comment by matthewturk** [comment]
@1mikegrn I think I might have been autocompleted by mistake - or I hope so!  This paper is news to me, I'm afraid.  :)

**Comment by dfm** [comment]
I think you mean @mdoucet!

@mdoucet: Can you let us know what sort of time frame you need to finish going through the checklist? Thanks!

**Comment by 1mikegrn** [comment]
@matthewturk haha ya, oops. Sorry about that!

**Comment by mdoucet** [comment]
@dfm Sorry for the delay. It's on my to-do list for this week.

**Comment by 1mikegrn** [comment]
@mdoucet no worries, appreciate the update :)

**Comment by dfm** [comment]
@mdoucet: Any updates on this? Please let me know if you have any questions!

**Comment by mdoucet** [comment]
I'm just done with my review. Sorry for the delay.

**Comment by dfm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01868/joss.01868/10.21105.joss.01868.pdf)

**Comment by dfm** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/j.jmat.2019.07.003 is OK
- 10.1016/j.nanoms.2019.02.001 is OK
- 10.1016/j.jmat.2018.12.005 is OK
- 10.1109/TMTT.1971.1127446 is OK
- 10.1016/j.matdes.2010.02.019 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by dfm** [comment]
@kjappelbaum, @mdoucet: I see that you have completed the checklist - thanks!! Do you have anything else you would like to add or any final recommendations?

@1mikegrn: Looking good! I can't seem to track down that Naito (1969) reference. Is it available online anywhere? Either way, can you now generate a new Zenodo release where the title and author list match this submission. Once you do that, can you post the new DOI and version number to this thread?

**Comment by dfm** [comment]
Also: can you all take a last quick look at the paper to make sure that the current version looks good?

**Comment by 1mikegrn** [comment]
Ya it took me about a month to track it down myself, ended up finding it though a university in Tel Aviv who sent me a pdf of their copy; it's the original paper the mathematics is legible, but the text is in Japanese.

[About the thickness of the ferrite absorption wall - Naito.pdf](https://github.com/openjournals/joss-reviews/files/3897903/About.the.thickness.of.the.ferrite.absorption.wall.-.Naito.pdf)

Also, what exactly needs to go in the Zenodo release? is it the entire package or just the .md file which points to the repository?

**Comment by dfm** [comment]
Thanks! The archive should be a snapshot of the full source. The easiest way to do it is following this guide: https://guides.github.com/activities/citable-code/

**Comment by 1mikegrn** [comment]
Thanks, that really helped :) 

Here's the DOI:

10.5281/zenodo.3555457

**Comment by dfm** [comment]
@whedon set 10.5281/zenodo.3555457 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3555457" target="_blank">10.5281/zenodo.3555457</a> is the archive.

**Comment by dfm** [comment]
@whedon set v1.1.0 as version

**Comment by whedon** [comment]
OK. v1.1.0 is the version.

**Comment by 1mikegrn** [comment]
Just a follow-up, but is there anything else that needs to be done before pushing this to production?

**Comment by danielskatz** [comment]
Once @dfm tells @openjournals/jose-eics that this is ready to publish, one of us (this week, me) will perform the final steps.

**Comment by dfm** [comment]
@kjappelbaum, @mdoucet: Can you please confirm that you are satisfied with the changes and recommend this submission for acceptance? Thanks!

**Comment by mdoucet** [comment]
@dfm all good!

**Comment by kjappelbaum** [comment]
@dfm I'm also fine with it.

**Comment by dfm** [comment]
@kjappelbaum, @mdoucet: Great. Thank you for your contributions to this review!!

**Comment by dfm** [comment]
Pinging @openjournals/jose-eics to say that this is ready for the final processing.

**Comment by dfm** [comment]
Pinging @openjournals/joss-eics to say that this is ready for the final processing.

**Comment by danielskatz** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01868/joss.01868/10.21105.joss.01868.pdf)

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/j.jmat.2019.07.003 is OK
- 10.1016/j.nanoms.2019.02.001 is OK
- 10.1016/j.jmat.2018.12.005 is OK
- 10.1109/TMTT.1971.1127446 is OK
- 10.1016/j.matdes.2010.02.019 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1150

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1150, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1201968380003201027) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1151
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01868](https://doi.org/10.21105/joss.01868)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by danielskatz** [comment]
Thanks to @kjappelbaum & @mdoucet for reviewing, and to @dfm for editing!

And congratulations to @1mikegrn and co-author!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01868/status.svg)](https://doi.org/10.21105/joss.01868)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01868">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01868/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01868/status.svg
   :target: https://doi.org/10.21105/joss.01868
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01868/status.svg)](https://doi.org/10.21105/joss.01868)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
